ci-operator/step-registry/ibmcloud/provision/bastionhost/ibmcloud-provision-bastionhost-commands.sh

```diff
@@ -0,0 +1,136 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+function run_command() {
+    local CMD="$1"
+    echo "Running Command: ${CMD}"
+    eval "${CMD}"
+}
+
+# IBM Cloud CLI login
+function ibmcloud_login {
+    "${IBMCLOUD_CLI}" login -a https://cloud.ibm.com -r ${LEASED_RESOURCE} --apikey @"${CLUSTER_PROFILE_DIR}/ibmcloud-api-key"
+}
+
+function checkCli() {
+  export IBMCLOUD_CLI=ibmcloud
+  export IBMCLOUD_HOME=/output
+  echo "check IBMCLOUD_CLI: ${IBMCLOUD_CLI}..."
+  command -v ${IBMCLOUD_CLI}
+  ${IBMCLOUD_CLI} --version
+  ${IBMCLOUD_CLI} plugin list
+}
+#####################################
+##############Initialize#############
+#####################################
+checkCli
+#create the bastion host based on the info (ibmcloud-provision-vpc ${SHARED_DIR}/customer_vpc_subnets.yaml), output ip info to ${SHARED_DIR}/bastion-info.yaml
+cluster_name="${NAMESPACE}-${JOB_NAME_HASH}"
+bastion_info_yaml="${SHARED_DIR}/bastion-info.yaml"
+
+bastion_name="${cluster_name}-bastion"
+MACHINE_TYPE="bx2-2x8"
+IMAGE="ibm-fedora-coreos-36-stable-2"
+workdir=${ARTIFACT_DIR}
+bastion_ignition_file="${SHARED_DIR}/${cluster_name}-bastion.ign"
+
+if [[ ! -f "${bastion_ignition_file}" ]]; then
+  echo "'${bastion_ignition_file}' not found, ignition-bastionhost step is required, abort." && exit 1
+fi
+
+VPC_CONFIG="${SHARED_DIR}/customer_vpc_subnets.yaml"
+if [[ ! -f "${VPC_CONFIG}" ]]; then
+  echo "Fail to find of VPC info file ${VPC_CONFIG}, abort." && exit 1
+fi
+echo "Reading variables from ${VPC_CONFIG}..."
+vpcName=$(yq-go r "${VPC_CONFIG}" 'platform.ibmcloud.vpcName')
+region=$(yq-go r "${VPC_CONFIG}" 'platform.ibmcloud.region')
+resource_group=$(yq-go r "${VPC_CONFIG}" 'platform.ibmcloud.resourceGroupName')
+echo "Using region: ${region}  resource_group: ${resource_group} vpc: ${vpcName}"
+
+# Disable exit on error for login
+set +o errexit
+login_success=false
+for _ in $(seq 5); do
```

Comment:
Even login also need retry here? Did you hit such issues in your testing?

Reply:
I have hit such issue on Prow CI

Reply:
Do you have several failed prow ci jobs log to let me have a check?